homeassistant/components/mqtt/__init__.py

```diff
@@ -9,7 +9,7 @@
 from operator import attrgetter
 import ssl
 import time
-from typing import Any, Callable, Union
+from typing import Any, Awaitable, Callable, Union, cast
```

Comment:
Hi There,

I think this line could perhaps change:

https://github.com/home-assistant/core/blob/9f43c994466f57bbc5b31856a8d2a09e858511a3/homeassistant/components/mqtt/__init__.py#L244

to 

SubscribePayloadType = str | bytes  # Only bytes if encoding is None


Should then be able to remove the Union import from typing if I'm not mistaken?

https://www.python.org/dev/peps/pep-0604/

Reply:
mypy doesn't accept a type alias like `SubscribePayloadType = str | bytes`